src/components/App.js

```diff
@@ -1,30 +1,57 @@
-import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, IconButton, VisibilityOff, FormControl, FormLabel, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
+import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, FormControl, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
 import AddIcon from '@mui/icons-material/Add'
-import Visibility from '@mui/icons-material/Visibility'
 import React, { useState } from "react"
+import { useAppData } from './context/hooks'
+import { MUTATION_KEYS, useMutation } from '../config/queryClient'
 
 function App() {
+  const { data } = useAppData()
+  const { mutate: postAppData } = useMutation(MUTATION_KEYS.POST_APP_DATA)
+
   const [question, setQuestion] = useState("");
-  const [choice1, setChoice1] = useState("");
-  const [choice2, setChoice2] = useState("");
+  const [choices, setChoices] = useState(["", ""]);
 
-  var choiceList = []
-  choiceList.push([choice1, setChoice1])
-  choiceList.push([choice2, setChoice2])
+  const [answers, setAnswers] = React.useState([{
+    answer: false
+  },
+  {
+    answer: false
+  }]);
 
-  //const [answers, setAnswers] = useState([{ value: 'it is answer 1', checked: false }])
+  const handleAnswerCorrectnessChange = (index, e) => {
+    let newAnswers = [...answers]
+    newAnswers[index].answer = e.target.checked
```

Comment:
I see a good attempt to not change the original array. However, this is not a deep copy so the original object will be modified despite this process!
The solution would be to change line 23 with:
```js
newAnswers[index] = { ...answers[index], answer: e.target.checked}
``` 
Or to use an immutable array.